tools/webpack/packages.js

```diff
@@ -0,0 +1,175 @@
+
+	const gutenbergPackages = [
+		'a11y',
+		'api-fetch',
+		'autop',
+		'blob',
+		'blocks',
+//		'block-library', // Not on npm yet.
+		'block-serialization-default-parser',
+//		'block-serialization-spec-parser', // No build-module folder.
+		'components',
+		'compose',
+		'core-data',
+		'data',
+		'date',
+		'deprecated',
+		'dom',
+		'dom-ready',
+//		'edit-post', // Not on npm yet.
+		'editor',
+		'element',
+//		'escape-html', // Not on npm yet.
+		'hooks',
+		'html-entities',
+		'i18n',
+//		'is-shallow-equal', // No build-module folder.
+		'keycodes',
+//		'list-reusable-blocks', // Not on npm yet.
+		'nux',
+		'plugins',
+		'redux-routine',
+//		'rich-text', // Not on npm yet.
+		'shortcode',
+		'token-list',
+		'url',
+		'viewport',
+		'wordcount',
+	];
+
+	const packagesStyles = [
+		'nux',
+		'components',
+		'editor',
+	];
+
+	const externals = {
+		react: 'React',
+		'react-dom': 'ReactDOM',
+		tinymce: 'tinymce',
+		moment: 'moment',
+		jquery: 'jQuery',
+		lodash: 'lodash',
+		'lodash-es': 'lodash',
+	};
+
+	gutenbergPackages.forEach( ( name ) => {
+		externals[ `@wordpress/${ name }` ] = {
+			this: [ 'wp', camelCaseDash( name ) ],
+		};
+	} );
+
+	const config = {
+		mode,
+
+		entry: gutenbergPackages.reduce( ( memo, packageName ) => {
+			const name = camelCaseDash( packageName );
+			memo[ name ] = `@wordpress/${ packageName }/build-module/index.js`;
+			return memo;
+		}, {} ),
+		output: {
+			filename: `[name]${ suffix }.js`,
+			path: join( baseDir, 'js/dist' ),
+			library: {
+				root: [ 'wp', '[name]' ]
+			},
+			libraryTarget: 'this',
+		},
+		externals,
+		resolve: {
+			modules: [
+				baseDir,
+				'node_modules',
+			],
+			alias: {
+				'lodash-es': 'lodash',
+			},
+		},
+		module: {
+			rules: [
+				{
+					test: /\.js$/,
+					use: [ 'source-map-loader' ],
+					enforce: 'pre',
+				},
+			],
+		},
+		plugins: [
+			new LibraryExportDefaultPlugin( [
+				'api-fetch',
+				'deprecated',
+				'dom-ready',
+				'redux-routine',
+			].map( camelCaseDash ) ),
+			new CopyWebpackPlugin(
+				packagesStyles.map( ( packageName ) => ( {
+					from: join( baseDir, `node_modules/@wordpress/${ packageName }/build-style/*.css` ),
+					to: join( baseDir, `styles/dist/${ packageName }/` ),
+					flatten: true,
+					transform: ( content ) => {
+						if ( config.mode === 'production' ) {
+							return postcss( [
+								require( 'cssnano' )( {
+									preset: 'default',
+								} ),
+							] )
+								.process( content, { from: 'src/app.css', to: 'dest/app.css' } )
+								.then( ( result ) => result.css );
+						}
+
+						return content;
+					}
+				} ) ),
+			),
+		],
+		stats: {
+			children: false,
+		},
+	};
+
+	if ( config.mode !== 'production' ) {
+		config.devtool = process.env.SOURCEMAP || 'source-map';
+	}
+
+	if ( config.mode === 'development' ) {
+		config.plugins.push( new LiveReloadPlugin( { port: process.env.GUTENBERG_LIVE_RELOAD_PORT || 35729 } ) );
+	}
+
+	return config;
+};
```

Comment:
I suppose we need also a babel config and a dedicated rule essentially to extract the i18n messages since the rest is already taken care of in the published packages.

Reply:
I will put extracting the i18n on my list as a follow-up task, I think we should get this in first though.

Reply:
I'm fine delaying it to a separate commit :)